File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxVal = 0
        index = 0
        count = 0
        while index < len(nums):
            if nums[index] == 0:
                maxVal = max(count, maxVal)
                count = 0
            else:
                count += 1
            index += 1
        maxVal = max(count, maxVal)
        return maxVal

File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int onesCount = 0;
        int result = 0;
        for(int i = 0; i < nums.length; i++) {
            if(nums[i] != 0) {
                onesCount += 1;
            } else {
                result = Math.max(result, onesCount);
                onesCount = 0;
            }
        }
        return Math.max(result, onesCount);
    }
}